module/localisation/BallLocalisation/src/BallLocalisation.cpp

```diff
@@ -155,22 +182,117 @@ namespace module::localisation {
+                        }
+
                         ball->time_of_measurement = last_time_update;
                         ball->Hcw                 = balls.Hcw;
+                        ball->average_rBWw        = field.Hfw.inverse() * average_rBFf;
                         if (log_level <= DEBUG) {
                             log<DEBUG>("rBWw: ", ball->rBWw.x(), ball->rBWw.y(), ball->rBWw.z());
                             log<DEBUG>("vBw: ", ball->vBw.x(), ball->vBw.y(), ball->vBw.z());
+                            log<DEBUG>("average rBWw: ",
+                                       ball->average_rBWw.x(),
+                                       ball->average_rBWw.y(),
+                                       ball->average_rBWw.z());
                             emit(graph("rBWw: ", ball->rBWw.x(), ball->rBWw.y(), ball->rBWw.z()));
                             emit(graph("vBw: ", ball->vBw.x(), ball->vBw.y(), ball->vBw.z()));
+                            log<DEBUG>("average_rBWw: ",
+                                       ball->average_rBWw.x(),
+                                       ball->average_rBWw.y(),
+                                       ball->average_rBWw.z());
                         }
 
                         emit(ball);
                     }
                 }
             });
+
+        // Stores ball positions received from teammates
+        on<Trigger<RoboCup>>().then([this](const RoboCup& robocup) {
+            if (cfg.use_r2r_balls) {
+                Eigen::Vector3d rBFf = robocup.ball.position.cast<double>();
+
+                team_guesses[robocup.current_pose.player_id - 1].last_heard = NUClear::clock::now();
+                team_guesses[robocup.current_pose.player_id - 1].rBFf       = rBFf;
+            }
+        });
+
+        // Called once a second to default to teammates balls if we haven't seen one recently
+        on<Every<1, Per<std::chrono::seconds>>, With<Field>>().then([this](const std::shared_ptr<const Field>& field) {
+            if (cfg.use_r2r_balls) {
+
+                const auto dt =
+                    std::chrono::duration_cast<std::chrono::duration<double>>(NUClear::clock::now() - last_time_update)
+                        .count();
+
+                if (dt > cfg.team_guess_default_timer) {
+
+                    Eigen::Vector3d average_rBFf = Eigen::Vector3d::Zero();
+                    get_average_team_rBFf(average_rBFf);
+
+                    average_rBFf = field->Hfw.inverse() * average_rBFf;
+
+                    last_time_update = NUClear::clock::now();
+                    auto ball        = std::make_unique<Ball>();
+
+                    ball->rBWw                = average_rBFf;
+                    ball->vBw                 = Eigen::Vector3d::Zero();
+                    ball->time_of_measurement = last_time_update;
+                    ball->Hcw                 = last_Hcw;
+                    emit(ball);
+                }
+            }
+        });
```

Comment:
I wonder if this check would be better in the RoboCup trigger. Then we're only checking if anyone is talked to us, and we update when theres new information.